### PR TITLE
feat(tests): add "AllSnapshots" binding context generator

### PR DESCRIPTION
**Pull request by diafour**
#### Overview

- Add "AllSnapshots" binding context generator.

- Replace custom binding context combiner for tests with CombineBindingContextForHook from ShellOperator.

#### Does this PR introduce a user-facing change?

```release-note

```

### CODE REVIEW

File: test/hook/context/context_combiner.go [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+type ContextCombiner struct {
+	op *shell_operator.ShellOperator
+	q  *queue.TaskQueue
+}
+
+func NewContextCombiner() *ContextCombiner {
+	op := &shell_operator.ShellOperator{}
+	op.TaskQueues = queue.NewTaskQueueSet()
+	op.TaskQueues.WithContext(context.Background())
+	op.TaskQueues.NewNamedQueue("main", nil)
```
Comment (nabokihms) [inline test/hook/context/context_combiner.go:27]: I did not get why there is the "main" queue now. Previously we did not specify any queue. And it is better to make `"main"` constant like `"TestTask"`.

File: test/hook/context/context_combiner.go [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+	t := task.NewTask(TestTaskType)
+	t.WithMetadata(task_metadata.HookMetadata{
+		Binding:        info.Binding,
+		Group:          info.Group,
+		BindingType:    bindingType,
+		BindingContext: info.BindingContext,
+		AllowFailure:   info.AllowFailure,
+	})
+	t.WithQueueName("main")
+	c.q.AddLast(t)
```
Comment (nabokihms) [inline test/hook/context/context_combiner.go:45]: Oh, we use queue only to add tasks? Ok then.

File: test/hook/context/context_combiner.go [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+// CombinedContext returns a combined context or a binding context
+// from the first task - as a Shell-operator will do.
+func (c *ContextCombiner) Combined() []binding_context.BindingContext {
+	firstTask := c.q.GetFirst()
+	if firstTask == nil {
+		return []binding_context.BindingContext{}
+	}
+	taskMeta := firstTask.GetMetadata()
+
+	bc := taskMeta.(task_metadata.BindingContextAccessor).GetBindingContext()
```
Comment (nabokihms) [inline test/hook/context/context_combiner.go:57]: Why do we need to cast a type here?

**Review by nabokihms — APPROVED** [review]
Approved in advance. The only concern is to move the default queue name to constant.

File: test/hook/context/context_combiner.go (hunk 3 above)
Comment (diafour) [reply test/hook/context/context_combiner.go:57]: Metadata is interface{} to have different HookMetadata in shell-operator and in addon-operator.

File: test/hook/context/context_combiner.go (hunk 2 above)
Comment (diafour) [reply test/hook/context/context_combiner.go:52]: I've added more comments and created a constant for queue's name.

**Pull request merged**